File: android-core/src/androidTest/java/com/mparticle/networking/PinningTestNetworkOptionsDisabled.java
package com.mparticle.networking;

import com.mparticle.MParticle;
import com.mparticle.MParticleOptions;

public class PinningTestNetworkOptionsDisabled extends PinningTest{

    @Override
    protected boolean shouldPin() {
        return true;
    }

    @Override
    protected MParticleOptions.Builder transformMParticleOptions(MParticleOptions.Builder builder) {
        return builder
                .environment(MParticle.Environment.Production)
                .networkOptions(NetworkOptions.builder()
                        .setPinningDisabledInDevelopment(true)
                        .build());
    }
}

File: android-core/src/androidTest/java/com/mparticle/networking/PinningTestNetworkOptionsDisabled.kt
package com.mparticle.networking

import com.mparticle.MParticle
import com.mparticle.toMParticleOptions

class PinningTestNetworkOptionsDisabled : PinningTest() {
    override fun shouldPin(): Boolean {
        return true
    }

    override fun transformMParticleOptions(builder: com.mparticle.api.MParticleOptions): com.mparticle.api.MParticleOptions {
        return builder.builder
            .environment(MParticle.Environment.Production)
            .networkOptions(
                NetworkOptions.builder()
                    .setPinningDisabledInDevelopment(true)
                    .build()
            )
            .toMParticleOptions()
    }
}
